include/portblas_helper.h

```diff
@@ -138,22 +138,24 @@ inline size_t get_num_compute_units(cl::sycl::queue &q) {
   @param size is the number of elements to be copied
 */
 template <typename element_t>
-inline cl::sycl::event copy_to_device(cl::sycl::queue q, const element_t *src,
-                                      BufferIterator<element_t> dst,
-                                      size_t size) {
+inline cl::sycl::event copy_to_device(
+    cl::sycl::queue q, const element_t *src, BufferIterator<element_t> dst,
+    size_t size, const std::vector<cl::sycl::event> &_dependencies = {}) {
   auto event = q.submit([&](cl::sycl::handler &cgh) {
     auto acc = dst.template get_range_accessor<cl::sycl::access::mode::write>(
         cgh, size);
+    cgh.depends_on(_dependencies);
```

Comment:
Is this dependency enforcement necessary? The buffer/accessor runtime should handle dependency and keep it correct, shouldn't it?

Reply:
Ideally speaking, this is not required for buffers since the dependencies are handled by the runtime. I added it here for completeness sake since we pass the dependencies to `queue.memcpy` for USM.

Reply:
I understand the completeness sake, but could this interfere with the runtime in some cases?

Reply:
I don't think so, no.